include/effects/matrix/PatternWeather.h

```diff
@@ -167,7 +167,7 @@ class PatternWeather : public LEDStripEffect
 
         if (httpResponseCode <= 0)
         {
-            debugW("Error fetching coordinates for location: %s", configLocation);
+            debugW("Error fetching coordinates for location: %s", configLocation.c_str());

```

Comment:
Is it worth adding stream operators so we can use String arguments naturally? You caught a lot of these, but it's an easy mistake to make.

As cringy as it is for a variadic macro, maybe we catch the case of the second argument being a String and just offer a specialzation for it. I wouldn't get crazy and do it for all arguments, but the debugX("blah %s blah", String) is such a common thing that a safety check - or a hard error if we can figure that out - may be worth it.

Reply:
For now I've enabled the warning that will result if you try to do this wrong!

Reply:
> As cringy as it is for a variadic macro, maybe we catch the case of the second argument being a String and just offer a specialzation for it.

I don't think we should, if I'm honest. If you do that for the "one string argument" case but not for the others, then it is even easier to forget the .c_str() for the second, third, ... String argument. Figuring out why the plain String does work for the first and not the others will then also be a very interesting exercise for the uninitiated - or the Rutger who comes back to his own code after a week or two.

> For now I've enabled the warning that will result if you try to do this wrong!

I'd say that is indeed the best thing to do for now.